ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1825,6 +1828,11 @@ CompletableFuture<Message> applyLogToStateMachine(LogEntryProto next) throws Raf
     return null;
   }
 
+  boolean isFirstLogAppliedByCurrentLeaderInCurrentTerm(long term) {
+    return term != state.getLastEntry().getTerm()
```

Comment:
Thanks for the patch! I guess you are trying to compare the term of the _current applying log_ with the _last applied log_ to decide the very first log of the current leader. However, `state.getLastEntry().getTerm()` returns the last **committed** log's term, not the last **applied** log's term.

Reply:
Yes, I made a mistake. Has fixed~